sdk/macro-frozen-abi/src/lib.rs

```diff
@@ -321,13 +321,18 @@ fn quote_for_test(
                     ::log::error!("digest error: {:#?}", result);
                 }
                 result.unwrap();
+                let actual_digest = format!("{}", hash);
                 if ::std::env::var("SOLANA_ABI_BULK_UPDATE").is_ok() {
-                    if #expected_digest != format!("{}", hash) {
+                    if #expected_digest != actual_digest {
                         #p!("sed -i -e 's/{}/{}/g' $(git grep --files-with-matches frozen_abi)", #expected_digest, hash);
                     }
                     ::log::warn!("Not testing the abi digest under SOLANA_ABI_BULK_UPDATE!");
                 } else {
-                  assert_eq!(#expected_digest, format!("{}", hash), "Possibly ABI changed? Confirm the diff by rerunning before and after this test failed with SOLANA_ABI_DUMP_DIR");
+                    if let Ok(dir) = ::std::env::var("SOLANA_ABI_DUMP_DIR") {
+                        assert_eq!(#expected_digest, actual_digest, "Possibly ABI changed? Examine the diff in SOLANA_ABI_DUMP_DIR!: $ diff -u {}/*{}* {}/*{}*", dir, #expected_digest, dir, actual_digest);
```

Comment:
@mvines I've made error message explicit how to proceed when encountering this. And, maintenance should be like this: aed8c474fc